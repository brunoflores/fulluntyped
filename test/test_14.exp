c: { get: Unit->Nat, inc: Unit->Unit }
unit: Unit
2: Nat
4: Nat
