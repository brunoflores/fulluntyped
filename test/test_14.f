c = let x = ref 1 in
      { get = (lambda _:Unit. !x),
        inc = (lambda _:Unit. x := succ (!x)) };

c.inc unit;

c.get unit;

(c.inc unit; c.inc unit; c.get unit);
