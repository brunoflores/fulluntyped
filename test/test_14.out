c: { get: Unit->Nat, inc: Unit->Unit }
